src/components/UniversalNavbar/UniversalNavbar.js

```diff
@@ -50,6 +50,15 @@ class UniversalNavbar extends React.Component {
   }
 
   render() {
+    const Link = ({ href, ...props }) => {
+      const PropsLink = this.props.link
+      if (PropsLink) {
+        return <PropsLink to={href} {...props} />
+      }
```

Comment:
created from props from UniversalNavbar

Reply:
I think is basically a good approach (wrap the injected component and route the props as needed), though I think it's nonperformance (at runtime) to create a new React component inside a render method, because each render the function will be recreated.

Reply:
nonperformant*